File: backend/library/src/main/java/com/pote/library/service/CategoryService.java
package com.pote.library.service;

import com.pote.library.dto.CategoryRequestDTO;
import com.pote.library.entity.Category;
import java.util.List;

public interface CategoryService {

	/**
	 * Create a new category (e.g., "Computer Science")
	 * 
	 * @param request
	 * 
	 *                Returns the Object of Category class
	 * @return
	 */
	Category addCategory(CategoryRequestDTO request);

	/**
	 * Get all categories
	 * 
	 * @return
	 */
	List<Category> getAllCategories();

	/**
	 * Get single category
	 * 
	 * @param id
	 * 
	 *           Returns the Object of Category class
	 * @return
	 */
	Category getCategoryById(Integer id);

	/**
	 * Delete a category Note:-> Only if no books are linked to it!
	 * 
	 * @param id
	 */
	void deleteCategory(Integer id);
}

File: frontend/library-Frontend/src/services/categoryService.js
import api from '../config/axiosConfig';

const categoryService = {
  // Get all categories
  getAll: async () => {
    const response = await api.get('/categories');
    return response.data;
  },

  // Create a new category
  create: async (categoryData) => {
    const response = await api.post('/categories', categoryData);
    return response.data;
  },

  // Delete a category
  delete: async (id) => {
    const response = await api.delete(`/categories/${id}`);
    return response.data;
  },
};

export default categoryService;
